src/clr-angular/forms/common/label.ts

```diff
@@ -3,22 +3,30 @@
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
-import {Directive, HostBinding, Input, OnDestroy, OnInit, Optional} from "@angular/core";
+import {Directive, ElementRef, HostBinding, Input, OnDestroy, OnInit, Optional, Renderer2} from "@angular/core";
 import {Subscription} from "rxjs";
 
-import {FormControlService} from "./form-control.service";
+import {IfErrorService} from "./if-error/if-error.service";
+import {ControlIdService} from "./providers/control-id.service";
 
 @Directive({selector: "label"})
 export class ClrLabel implements OnInit, OnDestroy {
-    constructor(@Optional() private formControlService: FormControlService) {}
+    labelClass: boolean;
+    constructor(@Optional() private controlIdService: ControlIdService,
+                @Optional() private controlService: IfErrorService, private renderer: Renderer2,
```

Comment:
Did you want to use the `NgControlService` here instead of the `IfErrorService`? I know it doesn't matter because the `input-container` declares both the services but the variable name you have chosen confused me.